File: src/main/java/com/rekeningrijden/simulation/Simulation/Gateway.java
package com.rekeningrijden.simulation.Simulation;

import com.rabbitmq.client.Channel;
import com.rabbitmq.client.Connection;
import com.rabbitmq.client.ConnectionFactory;
import org.apache.log4j.Logger;

public class Gateway {
    private final static Logger logger = Logger.getLogger(Gateway.class);

    private ConnectionFactory factory;
    public Connection connection;
    public Channel channel;

    public Gateway() {
        factory = new ConnectionFactory();
        factory.setHost("localhost");
    }

    public void connect() {
        try {
            connection = factory.newConnection();
            channel = connection.createChannel();
        }
        catch (Exception e) {
            logger.fatal("Connection with the message queue could not be established.");
            System.exit(1);
        }
    }
}

File: src/main/kotlin/com/rekeningrijden/simulation/simulation/Gateway.kt
package com.rekeningrijden.simulation.simulation

import com.rabbitmq.client.Channel
import com.rabbitmq.client.Connection
import com.rabbitmq.client.ConnectionFactory
import org.apache.log4j.Logger

class Gateway {
    private val factory: ConnectionFactory
    var connection: Connection? = null
    var channel: Channel? = null

    constructor(host: String) {
        factory = ConnectionFactory()
        factory.host = host
    }

    fun connect() {
        try {
            connection = factory.newConnection()
            channel = connection?.createChannel()
        } catch (e: Exception) {
            logger.fatal("Connection with the message queue could not be established.")
            logger.fatal(e.toString())
            System.exit(1)
        }
    }

    companion object {
        private val logger = Logger.getLogger(Gateway::class.java)
    }
}
